src/parser/avc_parser.cpp

```diff
@@ -192,6 +193,11 @@ ParserResult AvcVideoParser::ParsePictureData(const uint8_t *p_stream, uint32_t
 
                     // Start decode process
                     if (num_slices_ == 0) {
+                        if (p_slice_header->field_pic_flag) {
+                            second_field_ = (pic_count_ & 1) ? 1 : 0;
```

Comment:
second_field = (pic_count_ & 1); will be sufficient here. Also if we move line #199 to #196, else condition can be avoided

Reply:
Yes. :) Simplified line 197. 

We need to keep line 199 for frame pictures (set to 0). second_filed_ value is checked later.